ui/src/tests/client.test.ts

```diff
@@ -56,6 +77,12 @@ describe('Client', () => {
         expect(await innerText(page, $table.cell(2, Col.Name))).toBe('phone');
         expect(await innerText(page, $table.cell(3, Col.Name))).toBe('desktop app');
     });
+    it('updates client', async () => {
+        await updateClient(1, {name: 'firefox'});
+    });
+    it('has updated client name', async () => {
+        await hasClient('firefox', 1);
```

Comment:
```suggestion
        await hasClient('firefox', 1)();
```
`hasClient` has type `(name: string, row: number): (() => Promise<void>)` so hasClient has type of `() => Promise<void>`, which is never called, so the test did not do anything. This is why it did not detect that the client name is not changed.